### DISCUSSION TITLE
Checking for null for `FileTrigger`'s `onSelect()` callback

### DISCUSSION THREAD

**Discussion by denchen** [adobe/react-spectrum#6507] [Q&A] [answered]
For `<FileTrigger>`, the arguments for the callback `onSelect()` is typed as possibly being null

```typescript
/**
  * Handler when a user selects a file.
  */
onSelect?: (files: FileList | null) => void,
```

Under what condition would this be null? A failed upload? No files selected? I ask because in your documentation, you don't check against null:

https://react-spectrum.adobe.com/react-aria/FileTrigger.html#example

```tsx
function Example(){
  let [file, setFile] = React.useState(null);

  return (
    <>
      <FileTrigger
        onSelect={(e) => {
          let files = Array.from(e);
          let filenames = files.map((file) => file.name);
          setFile(filenames);
        }}>
        <Button>Select a file</Button>
      </FileTrigger>
      {file && file}
    </>
  )
}

**Comment by snowystinger** [comment, marked as answer]
The type for typescript `input onChange` always includes null, it doesn't know what type of event it is. I guess we could assert that it's never null for this one, as suggested in this discussion https://www.reddit.com/r/typescript/comments/19dqio6/what_is_the_correct_type_for_input_file_onchange/
